binderhub/build.py

```diff
@@ -394,7 +406,23 @@ def submit(self):
             )
         ]
 
-        if self.push_secret:
+        env = [
+            client.V1EnvVar(name=key, value=value)
+            for key, value in self.extra_envs.items()
+        ]
+        if self.git_credentials:
+            env.append(
+                client.V1EnvVar(name="GIT_CREDENTIAL_ENV", value=self.git_credentials)
+            )
+
+        if self.push_secret_content:
+            env.append(
+                client.V1EnvVar(
+                    name="CONTAINER_ENGINE_REGISTRY_CREDENTIALS",
+                    value=self.push_secret_content,
```

Comment:
This is less secure than the `push_secret` option I think, because `kubectl get pod` can reveal content while when using a dedicated k8s Secret to house the content only `kubectl get secret` can reveal the content.

KubeSpawner has ended up dynamically creating also a k8s Secret for user pods to house misc information, for example info relevant for internal_ssl. It certainly adds complexity to do that, and this is far less complicated.

I think it may be acceptable to do this if we describe it clearly in the help string of the config.

Reply:
I agree it's not as good as using a secret, however we had a long discussion in https://github.com/jupyterhub/binderhub/pull/1577 over the pros and cons of dynamically creating secrets attached to pods, as the same problem arises with dynamic Git credentials.